### PR TITLE
[REVIEW] Fix comments for make_dictionary_column factory functions

### LINKED ISSUE #4658: [DOC] Dictionary column factory method incorrectly specifies null mask behaviour
## Report incorrect documentation

**Location of incorrect documentation**
https://github.com/rapidsai/cudf/blob/1b9406816de3dbaf9e36811e5fd40d305b5fe087/cpp/include/cudf/dictionary/dictionary_factories.hpp#L72-L74

**Describe the problems or issues found in the documentation**
The documentation says the null mask is copied from the indices column whereas the very next line says indices cannot be nullable

**Steps taken to verify documentation is incorrect**
Read the implementation
https://github.com/rapidsai/cudf/blob/1b9406816de3dbaf9e36811e5fd40d305b5fe087/cpp/src/dictionary/dictionary_factories.cu#L61

**Suggested fix for documentation**
I'd prefer if the null mask was taken from indices column as it would simplify things. e.g. in testing, we wouldn't have to make a null mask separately. We could make a `fixed_width_column_wrapper<int32_t>` with nulls to construct a dictionary column.

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #4658 
Comment about the null-mask was on the wrong on both dictionary factory functions.
Also corrected some doxygen warnings and added comment to `strings` and `dictionary` namespaces so they appear nicely in the generated docs.

### CODE REVIEW

File: cpp/include/cudf/dictionary/dictionary_factories.hpp [hunk 1]
```diff
@@ -43,6 +43,8 @@ namespace cudf
  * d is now {["a","c","d"],[1,0,0,2,2]}
  * ```
  *
+ * The null_mask and null count for the output column are copied from the indices column.
```
Comment (jrhemstad) [inline cpp/include/cudf/dictionary/dictionary_factories.hpp]: Maybe add a null element to the example above.

File: cpp/include/cudf/dictionary/dictionary_factories.hpp [hunk 2]
```diff
@@ -69,8 +71,6 @@ std::unique_ptr<column> make_dictionary_column( column_view const& keys_column,
  *
  * The indices values must be in the range [0,keys_column.size()).
  *
- * The null_mask and null count for the output column are copied from the indices column.
```
Comment (jrhemstad) [inline cpp/include/cudf/dictionary/dictionary_factories.hpp]: It's potentially confusing what the difference is between this factory and the one above.

I'd change the brief to say that this factory doesn't make a copy and instead takes ownership of the columns.

```
 * @brief Construct a dictionary column by taking ownership of the provided keys
 * and indices columns. 
```

File: cpp/include/cudf/dictionary/dictionary_factories.hpp (hunk 1 above)
Comment (davidwendt) [reply cpp/include/cudf/dictionary/dictionary_factories.hpp]: What would that look like? Something like this?
```
 k = ["a","c","d"]
 i = [1,0,null,2,2]
 d = make_dictionary_column(k,i)
 d is now {["a","c","d"],[1,0,unk,2,2]} bitmask={1,1,0,1,1}
 ```
I don't want to imply the indices have a null bitmask.

Comment (jrhemstad) [reply cpp/include/cudf/dictionary/dictionary_factories.hpp]: sure, that looks good.

**Pull request merged**